mobile/lib/widgets/asset_grid/multiselect_grid.dart

```diff
@@ -204,11 +204,26 @@ class MultiselectGrid extends HookConsumerWidget {
     void onDeleteLocal(bool onlyBackedUp) async {
       processing.value = true;
       try {
-        final localIds = selection.value.where((a) => a.isLocal).toList();
+        // Filter local assets based on the backup status (`isRemote`).
+        // If `onlyBackedUp` is true, only select assets that are also present on the server (`isRemote`).
+        final localIds = selection.value
+            .where((a) => a.isLocal && (!onlyBackedUp || a.isRemote))
+            .toList();
 
+        if (localIds.isEmpty) {
+          ImmichToast.show(
+            context: context,
+            msg: 'No backed-up assets selected for deletion.',
+            gravity: ToastGravity.BOTTOM,
+          );
+          return;
+        }
+
+        // Proceed with deletion of filtered assets
         final isDeleted = await ref
             .read(assetProvider.notifier)
             .deleteLocalOnlyAssets(localIds, onlyBackedUp: onlyBackedUp);
```

Comment:
The `onlyBackup` parameter is already handled in `deleteLocalOnlyAssets` method. The fix to this issue should ideally be made inside the method.